src/EditorFeatures/Test/Diagnostics/TestDiagnosticAnalyzerService.cs

```diff
@@ -40,8 +42,9 @@ internal TestDiagnosticAnalyzerService(ImmutableArray<AnalyzerReference> hostAna
             HostAnalyzerManager hostAnalyzerManager,
             AbstractHostDiagnosticUpdateSource hostDiagnosticUpdateSource,
             Action<Exception, DiagnosticAnalyzer, Diagnostic> onAnalyzerException,
-            IDiagnosticUpdateSourceRegistrationService registrationService = null)
-            : base(hostAnalyzerManager, hostDiagnosticUpdateSource, registrationService ?? new MockDiagnosticUpdateSourceRegistrationService())
+            IDiagnosticUpdateSourceRegistrationService registrationService = null,
+            IAsynchronousOperationListener listener = null)
```

Comment:
well before this code, all consumer of TestDiagnosticAnalyzerService didn't need listener.